wkcuber/wkcuber/upload.py

```diff
@@ -1,72 +1,64 @@
-from argparse import ArgumentParser
-from os import environ
+"""This module takes care of downsampling WEBKNOSSOS datasets."""
+
+from pathlib import Path
+from typing import Annotated, Optional
+
+import typer
+from rich import print as rprint
 
 from webknossos import Dataset, webknossos_context
 from webknossos.client._defaults import DEFAULT_WEBKNOSSOS_URL
 from webknossos.client._upload_dataset import DEFAULT_SIMULTANEOUS_UPLOADS
 
-from ._internal.utils import (
-    add_verbose_flag,
-    parse_path,
-    setup_logging,
-    setup_warnings,
-    AsciiArgumentParser,
-)
-
-
-def create_parser() -> ArgumentParser:
-    parser = AsciiArgumentParser()
-
-    parser.add_argument(
-        "source_path",
-        help="Directory containing the source WKW dataset.",
-        type=parse_path,
-    )
-
-    parser.add_argument(
-        "--token",
-        help="Auth token of the user on WEBKNOSSOS",
-        default=None,
-    )
+app = typer.Typer(invoke_without_command=True)
 
-    parser.add_argument(
-        "--url", help="Base url of the WEBKNOSSOS instance", default=None
-    )
 
-    parser.add_argument(
-        "--jobs",
-        "-j",
-        default=DEFAULT_SIMULTANEOUS_UPLOADS,
-        type=int,
-        help=f"Number of simultaneous upload processes. Defaults to {DEFAULT_SIMULTANEOUS_UPLOADS}.",
-    )
+@app.callback()
+def main(
+    *,
+    path: Annotated[
+        Path,
+        typer.Argument(
+            show_default=False,
+            help="Path to your local WEBKNOSSOS dataset.",
+        ),
+    ],
+    url: Annotated[
+        str,
+        typer.Option(
+            help="URL to WEBKNOSSOS instance.",
+            rich_help_panel="WEBKNOSSOS context",
+            envvar="WK_URL",
+        ),
+    ] = DEFAULT_WEBKNOSSOS_URL,
+    token: Annotated[
+        Optional[str],
+        typer.Option(
+            help="Authentication token for WEBKNOSSOS instance.",
+            rich_help_panel="WEBKNOSSOS context",
+            envvar="WK_TOKEN",
+        ),
+    ] = None,
+    dataset_name: Annotated[
+        Optional[str],
+        typer.Option(
+            help="Name of your dataset on your WEBKNOSSOS.",
```

Comment:
Is this a renaming feature? Then I would use the words `rename`in the description, e.g. `Alternative name to rename your dataset on upload to WEBKNOSSOS`.